python/tvm/octo/compile.py

```diff
@@ -0,0 +1,157 @@
+    target : tvm.target.Target
+        The target used for compilation. Needed to parameterize CUTLASS.
+
+    Returns
+    -------
+    cutlass_mod : tvm.IRModule
+        The input graph with all possible subgraphs rewritten.
+    """
+    # Extract the sm version of the current target.
+    assert target.arch, "Target architecture must be specified."
+    sm = int(target.arch.split("_")[1])
+
+    # Construct CUTLASS codegen pass.
+    cutlass_codegen_pass = relax.transform.RunCodegen(
+        {"cutlass": {"sm": sm, "find_first_valid": True}}
+    )
+
+    # Construct pattern identification pass.
+    # TODO(jwfromm) rebase on cutlass pattern language
+
+    # Run passes on input module.
+    seq = tvm.transform.Sequential(
+        [
+            # relax.transform.FuseOpPattern(patterns, annotate_codegen=True),
+            cutlass_codegen_pass
+        ]
+    )
+
+    return seq(mod)
+
+
+def compile(
+    model: Union[str, Path, onnx.ModelProto],
+    target: Optional[tvm.target.Target] = None,
+    shape_dict: Optional[Dict[str, List]] = None,
+):
+    """Entrypoint to compiling a model using the Unity Flow.
+
+    Parameters
+    ----------
+    model : Union[str, Path, onnx.ModelProto]
+        An input onnx model to convert. Can either be a path to a model or an already
+        loaded onnx protobuf.
+
+    target : Optional[tvm.target.Target]
+        A description of the hardware to compile to. If not provided, one will be extracted for
+        the current host machine.
+
+    shape_dict : Optional[Dict[str, List]]
+        An optional dictionary that maps inputs to specific shapes. If not provided,
+        the default values in the onnx graph will be used.
+
+    Returns
+    -------
+    octo_model: OctoModel
+        A convenience wrapper around the compiled model that provides utility functions.
+    """
+    # Determine current target.
+    if target is None:
+        # Check if this is gpu enabled.
+        if tvm.cuda(0).exist:
+            target = get_cuda_target()
+        else:
+            target = get_llvm_target()
+        print("Compiling with target %s" % str(target))
+
+    # Convert model into a relax module.
+    relax_mod = load_onnx_model(model, shape_dict)
+
+    # Extract information about input shapes and types so we can
+    # randomly generate them later if needed.
+    input_info = {}
+    for inp in relax_mod["main"].params:
+        input_shape = [i.value for i in inp.struct_info.shape]
+        input_dtype = inp.struct_info.dtype
+        input_info[inp.name_hint] = (input_shape, input_dtype)
+
+    # Match subgraphs that can be offloaded to cutlass and offload them.
+    # TODO(jwfromm) Currently doesnt work, get one e2e example.
+    # offload_cutlass(relax_mod, target)
+
+    # Perform legalization to lower Relax operators.
+    relax_mod = relax.transform.LegalizeOps()(relax_mod)
+
+    # Schedule all remaining functions to be compatible with gpu if needed.
+    if str(target.kind) == "cuda":
```

Comment:
is str needed?

Reply:
Yeah, the default type of `target.kind` is `TargetKind`.

Reply:
we can use `target.kind.name` to compare